### DISCUSSION TITLE
Multiple Worlds support?

### DISCUSSION THREAD

**Discussion by Ggcpp** [Trouv/bevy_ecs_ldtk#315] [Q&A]
Hi!!

Does the plugin support multiple worlds in one project? (LDtk advanced settings)
The book mentions that it is possible to have multiple ```LdtkWorldBundle``` spawned so I guess it is possible to have multiple worlds but I don't know how.

**Comment by Trouv** [comment]
There is some support of multiple worlds, but there aren't any features built around it. It's more like, you should be able to use the plugin normally if you have multiple worlds.

Actually, you should just be able to use one "`LdtkWorldBundle`" to spawn multiple worlds. This bundle was given its name before multiple worlds was really a thing, so it could just as easily be called `LdtkProjectBundle`. It's really all about level selection, you should be able to just choose the levels to spawn in any world at any time. I know from your other post you have some custom level selection going on, but the only main difference for using `LevelSelection` to do so is in its `Indices` variant. There is a convenience method [`LevelSelection::index`](https://docs.rs/bevy_ecs_ldtk/latest/bevy_ecs_ldtk/prelude/enum.LevelSelection.html#method.index) to create an `Indices` variant for users that only have one world, but you can also use this variant to supply both a world index and a level index via [`LevelSelection::indices`](https://docs.rs/bevy_ecs_ldtk/latest/bevy_ecs_ldtk/prelude/enum.LevelSelection.html#method.indices). Of course, you can use any other level selection mechanisms as well, I'm just going into more detail about this one since it is special in the multiple-worlds case.

If you'd rather organize the levels you spawn in different `LdtkWorldBundle`s per world, I think that's a fine strategy. However, you should know that things that depend on the hierarchy of the world bundle won't necessarily translate from world to world. For example [Worldly Entities](https://trouv.github.io/bevy_ecs_ldtk/latest/explanation/anatomy-of-the-world.html#worldly-entities) will still only really belong to the world bundle they spawned in. If their birth world despawns, they will too, assuming you haven't intervened somehow.
